### ISSUE TITLE
read_text returns empty list for an empty file

### ISSUE THREAD

**Issue by mattlaw** [Leont/file-slurp-sane#11]
Calling `read_text` in list context normally returns a scalar as expected, unless the file in question is empty.

```
$ perl -MFile::Slurper=read_text -E 'my @foo = (read_text("/dev/null")); say 0+@foo'
0
```

So a statement like `localise(read_text($file, "UTF-8"), $locale)` passes `$locale` as the first argument if and only if `$file` is empty, assuming no prototype on the sub.

The documentation strongly implies that the return value is always a scalar 

> Reads file `$filename` into a scalar

**Comment by Leont** [comment]
You're right, it should return an empty string in such a case.

**Comment by Leont** [comment]
Released 0.014 with a fix
